client/my-sites/checkout/src/lib/pending-page.ts

```diff
@@ -335,7 +345,7 @@ export function getRedirectFromPendingPage( {
 					redirectTo ?? getDefaultSuccessUrl( siteSlug, receiptId ),
 					receiptId
 				),
-				siteSlug,
+				siteSlug || fromSite,
```

Comment:
I don't fully understand the use of `fromSite` but should this be added to the other return values of this function as well? Most purchases don't have a receipt ID at this stage and will end up in the `if ( transaction )` block below.

Reply:
Hmm, yes maybe so..  I mean, it has been working when the `fromSiteSlug` was only in the `if ( receiptId )` block and not the others, but it does make sense that it should probably be added to the `if ( transaction )` block also. So I'm adding it. 👍
All I know is that `fromSiteSlug` should be available here to complete the transaction during the siteless checkout flow (`/checkout/jetpack/:product_slug`) when the following 3 url query args are passed from the site -> connectAfterCheckout(true), fromSiteSlug, and adminUrl.

Basically, 'fromSiteSlug' is the "site-slug passed from the site" via url query arg when we do not has the site in context (ie- the site is not connected). 
 As opposed to `siteSlug` which can be retrieved only when the site is in context (which we don't have in the siteless checkout flow).

Reply:
I suggest testing the flow with PayPal as the payment method if you haven't already. That flow is different enough from the others (it requires like 4 redirects altogether) that it can highlight bugs where query parameters or state variables might be lost.

Reply:
I updated the PR so that this "connect after checkout" flow works correctly with PayPal and generic payment processors. 👍